app.js

```diff
@@ -1,8 +1,127 @@
+const optionBEl = document.getElementById('option-b');
+const optionBSubtractButton = document.getElementById('option-b-subtract');
+const optionBVoteCountEl = document.getElementById('option-b-vote-count');
+const optionBAddButton = document.getElementById('option-b-add');
+
+const closePollButton = document.getElementById('close-poll-button');
+
+const pastPollsDivEl = document.getElementById('past-polls-div');
 
 // let state
+let currentPoll = {
+    question: '',
+    optionA: '',
+    optionB: '',
+    votesA: 0,
+    votesB: 0
+};
+
+const pastPollsArray = [];
 
 // set event listeners 
-  // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+optionASubtractButton.addEventListener('click', () => {
+    currentPoll.votesA--;
+    optionAVoteCountEl.textContent = currentPoll.votesA;
+});
+
+optionAAddButton.addEventListener('click', () => {
+    currentPoll.votesA++;
+    optionAVoteCountEl.textContent = currentPoll.votesA;
+});
+
+optionBSubtractButton.addEventListener('click', () => {
+    currentPoll.votesB--;
+    optionBVoteCountEl.textContent = currentPoll.votesB;
+});
+
+optionBAddButton.addEventListener('click', () => {
+    currentPoll.votesB++;
+    optionBVoteCountEl.textContent = currentPoll.votesB;
+});
+
+newPollFormEl.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const formData = new FormData(newPollFormEl);
+
+    // get user input
+    const userQuestion = formData.get('user-question-input');
+    const userOptionA = formData.get('option-a-input');
+    const userOptionB = formData.get('option-b-input');
+
+    // use user input to update state
+    currentPoll.question = userQuestion;
+    currentPoll.optionA = userOptionA;
+    currentPoll.optionB = userOptionB;
+
+    newPollFormEl.reset();
+
+    // update DOM to reflect the new state
+    displayCurrentPoll();
+});
+
+closePollButton.addEventListener('click', () => {
+  
+    const poll = renderPoll();
+  
+    pastPollsArray.push(poll);
+
+    resetState();
+
+    displayAllPolls();
+});
+  
+function displayCurrentPoll() {
+    currentPollQuestion.textContent = currentPoll.question;
+    optionAEl.textContent = currentPoll.optionA;
+    optionBEl.textContent = currentPoll.optionB;
+    optionAVoteCountEl.textContent = currentPoll.votesA;
+    optionBVoteCountEl.textContent = currentPoll.votesB;
+}
+
+function resetState() {
+    currentPoll = {
+        question: '',
+        optionA: '',
+        optionB: '',
+        votesA: 0,
+        votesB: 0,
+    };
+
+    currentPollQuestion.textContent = '-';
+    optionAEl.textContent = '';
+    optionBEl.textContent = '';
+    optionAVoteCountEl.textContent = '0';
+    optionBVoteCountEl.textContent = '0';
+}
+
+function renderPoll() {
+    return {
+        question: currentPoll.question,
+        optionA: currentPoll.optionA,
+        optionB: currentPoll.optionB,
+        votesA: currentPoll.votesA,
+        votesB: currentPoll.votesB
+    };
+}
```

Comment:
This isn't necessary since the only thing it's doing is returning an object that is identical to what is already stored in `currentPoll` at this point. It is creating and returning a new object, but why?